stdlib/tkinter/filedialog.pyi

```diff
@@ -57,11 +58,76 @@ class SaveAs(_Dialog):
 class Directory(commondialog.Dialog):
     command: ClassVar[str] = ...
 
-def askopenfilename(**options): ...
-def asksaveasfilename(**options): ...
-def askopenfilenames(**options): ...
-def askopenfile(mode: str = ..., **options): ...
-def askopenfiles(mode: str = ..., **options): ...
-def asksaveasfile(mode: str = ..., **options): ...
-def askdirectory(**options): ...
+# TODO: command kwarg available on macos
+def asksaveasfilename(
+    *,
+    confirmoverwrite: bool | None = ...,
+    defaultextension: str | None = ...,
+    filetypes: Iterable[Tuple[str, str] | Tuple[str, _TkinterSequence[str]]] | None = ...,
+    initialdir: StrOrBytesPath | None = ...,
+    initialfile: StrOrBytesPath | None = ...,
+    parent: Misc | None = ...,
+    title: str | None = ...,
+    typevariable: StringVar | str | None = ...,
+) -> str: ...  # can be empty string
+def askopenfilename(
+    *,
+    defaultextension: str | None = ...,
+    filetypes: Iterable[Tuple[str, str] | Tuple[str, _TkinterSequence[str]]] | None = ...,
+    initialdir: StrOrBytesPath | None = ...,
+    initialfile: StrOrBytesPath | None = ...,
+    parent: Misc | None = ...,
+    title: str | None = ...,
+    typevariable: StringVar | str | None = ...,
+) -> str: ...  # can be empty string
+def askopenfilenames(
+    *,
+    defaultextension: str | None = ...,
+    filetypes: Iterable[Tuple[str, str] | Tuple[str, _TkinterSequence[str]]] | None = ...,
+    initialdir: StrOrBytesPath | None = ...,
+    initialfile: StrOrBytesPath | None = ...,
+    parent: Misc | None = ...,
+    title: str | None = ...,
+    typevariable: StringVar | str | None = ...,
+) -> Literal[""] | Tuple[str, ...]: ...
+def askdirectory(
+    *, initialdir: StrOrBytesPath | None = ..., mustexist: bool | None = ..., parent: Misc | None = ..., title: str | None = ...
+) -> str: ...  # can be empty string
+
+# TODO: If someone actually uses these, overload to have the actual return type of open(..., mode)
```

Comment:
During several years of tkinter, I haven't seen anyone actually using these functions. They ask the file name, just like `askopenfilename()` or `asksaveasfilename()` or `askopenfilenames()` would, and pass it to the built-in `open()` function. When using these functions, it's not possible to pass other arguments to `open()`, such as `encoding` or `errors`.